Sources/Foundation/JSONEncoder.swift

```diff
@@ -237,176 +237,405 @@ open class JSONEncoder {
+            if case .convertToString(let posInfString, let negInfString, let nanString) = self.options.nonConformingFloatEncodingStrategy {
+                switch float {
+                case F.infinity:
+                    return .string(posInfString)
+                case -F.infinity:
+                    return .string(negInfString)
+                default:
+                    // must be nan in this case
+                    return .string(nanString)
+                }
+            }
+            
+            var path = self.codingPath
+            if let additionalKey = additionalKey {
+                path.append(additionalKey)
+            }
+            
+            throw EncodingError.invalidValue(float, .init(
+                codingPath: path,
+                debugDescription: "Unable to encode \(F.self).\(float) directly in JSON."
+            ))
+        }
+        
+        var string = float.description
+        if string.hasSuffix(".0") {
+            string.removeLast(2)
+        }
+        return .number(string)
+    }
+    
+    fileprivate func wrapEncodable<E: Encodable>(_ encodable: E, for additionalKey: CodingKey?) throws -> JSONValue? {
+        switch encodable {
+        case let date as Date:
+            return try self.wrapDate(date, for: additionalKey)
+        case let data as Data:
+            return try self.wrapData(data, for: additionalKey)
+        case let url as URL:
+            return .string(url.absoluteString)
+        case let decimal as Decimal:
+            return .number(decimal.description)
+        case let object as [String: Encodable]:
+            return try self.wrapObject(object, for: additionalKey)
+        default:
+            let encoder = self.getEncoder(for: additionalKey)
+            try encodable.encode(to: encoder)
+            return encoder.value
+        }
+    }
+    
+    func wrapDate(_ date: Date, for additionalKey: CodingKey?) throws -> JSONValue {
+        switch self.options.dateEncodingStrategy {
+        case .deferredToDate:
+            let encoder = self.getEncoder(for: additionalKey)
+            try date.encode(to: encoder)
+            return encoder.value ?? .null
 
-    /// The container stack.
-    /// Elements may be any one of the JSON types (NSNull, NSNumber, NSString, NSArray, NSDictionary).
-    private(set) fileprivate var containers: [NSObject] = []
+        case .secondsSince1970:
+            return .number(date.timeIntervalSince1970.description)
 
-    // MARK: - Initialization
+        case .millisecondsSince1970:
+            return .number((date.timeIntervalSince1970 * 1000).description)
 
-    /// Initializes `self` with no containers.
-    fileprivate init() {}
+        case .iso8601:
+            if #available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *) {
+                return .string(_iso8601Formatter.string(from: date))
+            } else {
+                fatalError("ISO8601DateFormatter is unavailable on this platform.")
```

Comment:
hmmm I think we can use regular formatter with right pattern to achieve iso8601, no?

Reply:
also afaik, iso8601 dateEncodingStrategy I supported now on linux

Reply:
I see it now